From e69758260b0403d47ec61692e1fc18bcbb14865e Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Fri, 18 May 2012 11:31:24 -0700
Subject: [PATCH] darwin: Address a build failure on Leopard and earlier OS
 versions

<https://trac.macports.org/ticket/34499>

Regression-from: 51691f0767f6a75a1f549cd979a878a0ad12a228
Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
(cherry picked from commit 27b821bc95ea01cc4292ada9c7c65211580d1f98)
---
 src/glx/apple/apple_glx_log.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/glx/apple/apple_glx_log.c b/src/glx/apple/apple_glx_log.c
index 9ebf666..5b9a865 100644
--- a/src/glx/apple/apple_glx_log.c
+++ b/src/glx/apple/apple_glx_log.c
@@ -76,7 +76,17 @@ void _apple_glx_vlog(int level, const char *file, const char *function,
     uint64_t thread = 0;
 
     if (pthread_is_threaded_np()) {
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1060
+        thread = (uint64_t)(uintptr_t)pthread_self();
+#elif MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+        if (&pthread_threadid_np) {
+            pthread_threadid_np(NULL, &thread);
+        } else {
+            thread = (uint64_t)(uintptr_t)pthread_self();
+        }
+#else
         pthread_threadid_np(NULL, &thread);
+#endif
     }
 
     if (diagnostic) {
-- 
1.7.10.1

